proposals/lip-0057.md

```diff
@@ -1343,23 +1343,25 @@ def getActiveValidators(validatorsTwoRoundsAgo: list[EligibleValidatorObject], r
         nbrInitValidators = initRounds + NUMBER_ACTIVE_VALIDATORS - roundNumber
         nbrElectedValidators = NUMBER_ACTIVE_VALIDATORS - nbrInitValidators
 
-        # In the definition below, recall that validatorsTwoRoundsAgo is sorted by validator weight.
-        electedValidatorAddresses = [item.address for item in validatorsTwoRoundsAgo[:nbrElectedValidators]]
-
-        remainingInitValidators = [address for address in initValidators if address not in electedValidatorAddresses]
-
-        # Concatenation of elected validators and remaining initial validators.
-        activeValidatorAddresses = electedValidatorAddresses + remainingInitValidators[:nbrInitValidators]
-        activeValidators = [{"address": item, "bftWeight": 1 } for item in activeValidatorAddresses]
-
+        # Recall that validatorsTwoRoundsAgo is sorted by validator weight.
+        electedValidators = validatorsTwoRoundsAgo[:nbrElectedValidators]
+        electedValidatorAddresses = [item.address for item in electedValidators]
+
+        chosenInitValidators = [address for address in initValidators if address not in electedValidatorAddresses][:nbrInitValidators]   
+        
+        # Determine active validators and their BFT weight.
+        activeValidators = [{"address": item.address, "bftWeight": ceiling(item.weight, WEIGHT_SCALE_FACTOR) } for item in electedValidators]
+        averageElectedValidatorWeight = sum([item.weight for item in electedValidators]) // nbrElectedValidators
+        activeValidators+= [{"address": item, "bftWeight": ceiling(averageElectedValidatorWeight, WEIGHT_SCALE_FACTOR)  } for item in chosenInitValidators]
```

Comment:
Here we shouldn't divide again by `WEIGHT_SCALE_FACTOR` as the average is taken already after the BFT weights have been divided by `WEIGHT_SCALE_FACTOR`.